### PR TITLE
Deprecate sim-based object access

### PR DESCRIPTION
**Pull request by jturner65**
## Motivation and Context
This PR changes all the existing tutorials and tests to access physical objects directly, [using the new managed rigid object functionality.](https://github.com/facebookresearch/habitat-sim/pull/1182).  The simulator bindings for the object_id-based access have also received a deprecation warning, indicating that they may be removed for Habitat-2.0.  The existing rigid-object-tutorial.py has been removed, as has its documentation, since this has been supplanted by the managed-rigid-object-tutorial.py

TODO: Object access in habitat-lab tutorials will also need to be changed to the new mechanism before the deprecated functionality is removed.

## How Has This Been Tested
All c++ and python tests pass

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: tests/test_physics.py [hunk 1]
```diff
@@ -413,7 +421,7 @@ def test_raycast():
```
Comment (eundersander) [inline tests/test_physics.py:421]: Comment on the line above: `assert raycast_results.hits[0].object_id == 0`

As-is, this code makes the dangerous assumption that the first created object gets id 0. The code is also not very readable. We should fix this line to be `== cube_obj.object_id`.

**Review by eundersander — APPROVED** [review]
I left some comments. LGTM.

File: examples/demo_runner.py [hunk 2]
```diff
@@ -208,6 +212,9 @@ def init_physics_test_scene(self, num_objects):
             )
 
     def do_time_steps(self):
+        # get the rigid object manager, which provides direct
```
[the remark this answers is not in the document]
Comment (jturner65) [reply examples/demo_runner.py:215]: I agree; my rationale is it's hard to tell where the first place someone is going to see this construct is, so I'm a bit redundant in order to catch everyone.  If you think this is over-done, I can prune them down.

File: examples/demo_runner.py [hunk 3]
```diff
@@ -186,19 +190,19 @@ def init_physics_test_scene(self, num_objects):
                     random.randint(-object_init_grid_dim[1], object_init_grid_dim[1]),
                     random.randint(-object_init_grid_dim[2], object_init_grid_dim[2]),
                 )
-            object_id = self._sim.add_object(rand_obj_index)
-            object_init_grid[object_init_cell] = object_id
+            obj = rigid_obj_mgr.add_object_by_id(rand_obj_index)
```
[the remark this answers is not in the document]
Comment (jturner65) [reply examples/demo_runner.py:193]: Good idea.

**Review by aclegg3 — APPROVED** [review]

File: examples/tutorials/nb_python/ECCV_2020_Interactivity.py [hunk 4]
```diff
@@ -846,22 +840,20 @@ def build_widget_ui(obj_attr_mgr, prim_attr_mgr):
...
 # @markdown Initial linear velocity (m/sec):
 lin_vel_x = 0  # @param {type:"slider", min:-10, max:10, step:0.1}
 lin_vel_y = 1  # @param {type:"slider", min:-10, max:10, step:0.1}
 lin_vel_z = 5  # @param {type:"slider", min:0, max:10, step:0.1}
-initial_linear_velocity = mn.Vector3(lin_vel_x, lin_vel_y, lin_vel_z)
-sim.set_linear_velocity(initial_linear_velocity, ball_id)
+ball_obj.linear_velocity = mn.Vector3(lin_vel_x, lin_vel_y, lin_vel_z)
 
 # @markdown Initial angular velocity (rad/sec):
 ang_vel_x = 0  # @param {type:"slider", min:-100, max:100, step:0.1}
```
Comment (Skylion007) [inline examples/tutorials/nb_python/ECCV_2020_Interactivity.py:853]: If these are floats, the Vector3d is unnecessary, right?

File: examples/tutorials/physics_benchmarking.py [hunk 5]
```diff
@@ -14,12 +14,7 @@
 output_path = os.path.join(dir_path, "physics_benchmarking_output/")
 
 
-# Define Helper Functions (Taken from rigid_object_tutorial.py)
```
Comment (Skylion007) [inline examples/tutorials/physics_benchmarking.py:17]: Comment needs to g.

File: examples/tutorials/nb_python/ECCV_2020_Interactivity.py (hunk 4 above)
Comment (jturner65) [reply examples/tutorials/nb_python/ECCV_2020_Interactivity.py:853]: Sometimes they get rounded to an int by the interface (when the sliders are moved), which is why they were built the way they were previously.

**Pull request merged**